### PR TITLE
[SPARK-40578][PS] Fix `IndexesTest.test_to_frame` when pandas 1.5.0

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fix the test `IndexesTest.test_to_frame` to support pandas 1.5.0

### Why are the changes needed?

The behavior of `Index.to_frame` is changed from pandas 1.5.0, but we cannot follow the behavior since we have a limitation for mixed type.

So we should create expected DataFrame manually to make the test pass.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UT fixed, and manually tested with pandas 1.5.0.

### CODE REVIEW

File: python/pyspark/pandas/tests/indexes/test_base.py [hunk 1]
```diff
@@ -203,9 +203,35 @@ def test_to_frame(self):
...
-            psidx.to_frame(name=[("x", 10), ("y", 20)]), pidx.to_frame(name=[("x", 10), ("y", 20)])
-        )
+        if LooseVersion(pd.__version__) < LooseVersion("1.5.0"):
+            self.assert_eq(
+                psidx.to_frame(name=[("x", 10), ("y", 20)]),
+                pidx.to_frame(name=[("x", 10), ("y", 20)]),
+            )
+        else:
+            # Since pandas 1.5.0, the result is changed as below:
+            #      (x, 10)  (y, 20)
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/tests/indexes/test_base.py:213]: What's the exact diff between <1.5 and >=1.5?

Comment (HyukjinKwon) [reply python/pyspark/pandas/tests/indexes/test_base.py:213]: mixed type in its name? or value? what's the mixed type here?

File: python/pyspark/pandas/tests/indexes/test_base.py [hunk 2]
```diff
@@ -203,9 +203,35 @@ def test_to_frame(self):
...
+            # 1 5        1        5
+            # 3 6        3        6
+            # 5 3        5        3
+            # 6 2        6        2
+            # 8 1        8        1
+            # 9 0        9        0
+            #   0        9        0
+            #   0        9        0
+            #
+            # pandas API on Spark doesn't support mixed type for Index,
```
Comment (zhengruifeng) [inline python/pyspark/pandas/tests/indexes/test_base.py:225]: `mixed type` means a `MultiIndex` with different atomic Index type on each level?

File: python/pyspark/pandas/tests/indexes/test_base.py (hunk 1 above)
Comment (itholic) [reply python/pyspark/pandas/tests/indexes/test_base.py:213]: The exact diff is shown as below:

```python
# pandas < 1.5.0
>>> pd.Index([('x', 10), ('y', 20)]).to_frame(name=[("x", 20), ("y", 20)])
      x   y
     20  20
x 10  x  10
y 20  y  20

# pandas >= 1.5.0
>>> pd.Index([('x', 10), ('y', 20)]).to_frame(name=[("x", 20), ("y", 20)])
     (x, 20)  (y, 20)
x 10       x       10
y 20       y       20
```

The column should be like `Index([('x', 20), ('y', 20)], dtype='object')`, but such as way is not allowed in pandas API on Spark:

```python
>>> ps.Index([['x', 20], ['y', 20]])
Traceback (most recent call last):
...
pyarrow.lib.ArrowTypeError: Expected bytes, got a 'int' object
```

Comment (itholic) [reply python/pyspark/pandas/tests/indexes/test_base.py:213]: Let me fix the comment to more explicitly.

File: python/pyspark/pandas/tests/indexes/test_base.py (hunk 2 above)
Comment (itholic) [reply python/pyspark/pandas/tests/indexes/test_base.py]: Could you refer to https://github.com/apache/spark/pull/38016#discussion_r980971632 ?

Just refine the comment, so please let me know if there is anything doesn't make sense!

**Review by HyukjinKwon — APPROVED** [review]

**Review by xinrong-meng — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**